pandas/core/nanops.py

```diff
@@ -384,8 +384,7 @@ def _na_for_min_count(
     else:
         assert axis is not None  # assertion to make mypy happy
         result_shape = values.shape[:axis] + values.shape[axis + 1 :]
-        result = np.empty(result_shape, dtype=values.dtype)
-        result.fill(fill_value)
+        result = np.full(result_shape, fill_value)
```

Comment:
does this not have a dtype parameter? does it matter?

Reply:
The result parameter here is what is making all the trouble. We set the dtype to datetime but our `NaT` is treated as `NaN` which numpy internally tries to cast into an integer which yields the error: 
`ValueError: cannot convert float NaN to integer`.
The conversion happens somewhere inside the c-part of numpy, which I'm not familiar with.

But to come back to your question: for our result, specifying the dtype is not necessary. The given outputs are all corresponding to the type of the input. But good point, I should probably add this check to my tests.